### ISSUE TITLE
MiKTeX console crash while changing remote package repository

### ISSUE THREAD

**Issue by balsigergil** [MiKTeX/miktex#867]
- [x] I have read and acknowledged this HOWTO article:  
  [Submitting a MiKTeX problem report](https://miktex.org/howto/bug-report)

Hi, I noticed that the MiKTeX console crashes when I try to change the remote package repository from the settings.

Here is my system config :
```
ReportDate: 2021-07-08 18:05:51
CurrentVersion: 21.6.28
SetupDate: 2021-07-07 14:25:00
SetupVersion: 21.6
Configuration: Regular
GitInfo: b96a3a3 / 2021-06-26 19:20:18
OS: Windows 10 Home, 64-bit, build 19042
SharedSetup: no
LinkTargetDirectory: C:\Users\***\AppData\Local\Programs\MiKTeX\miktex\bin\x64
PathOkay: yes
LastUpdateCheck: 2021-07-07 14:25:21
LastUpdate: 2021-07-07 14:25:58
LastUpdateDb: 2021-07-07 14:25:20
SystemAdmin: yes
RootPrivileges: no
AdminMode: no
Root0: C:\Users\***\AppData\Roaming\MiKTeX
Root1: C:\Users\***\AppData\Local\MiKTeX
Root2: C:\Users\***\AppData\Local\Programs\MiKTeX
UserInstall: C:\Users\***\AppData\Local\Programs\MiKTeX
UserConfig: C:\Users\***\AppData\Roaming\MiKTeX
UserData: C:\Users\***\AppData\Local\MiKTeX
```

I did some investigations and I found that the mirror `https://mirror.foobar.to/` is down. I also noticed that the console do a request to https://api2.miktex.org/hello but always return status `429 Too Many Requests`. I don't know if it's normal but it seems strange.

Here is the miktex-console logs :
```
2021-07-08 18:07:07,631+0200 INFO  main - starting: MiKTeX Console 4.2.1 (MiKTeX 21.6.28)
2021-07-08 18:07:15,697+0200 INFO  miktex-console.packagemanager - going to download https://api2.miktex.org/hello
2021-07-08 18:07:15,861+0200 INFO  miktex-console.packagemanager - going to download https://api2.miktex.org/repositories?&releaseState=Stable
```

**Comment by edocevoli** [comment]
I can reproduce the bug. There are two workarounds for now:

1. choose a random package repository
2. set a package repository (you must know its URL):
    `mpm --set-repository=https://mirror.foobar.to/CTAN/systems/win32/miktex/tm/packages/`

**Comment by edocevoli** [comment]
This has been fixed in 21.7.
